test/targets.cpp

```diff
@@ -50,4 +53,21 @@ TEST_CASE(targets)
     EXPECT(ts.size() >= 1);
 }
 
+TEST_CASE(concurrent_targets)
```

Comment:
This test does not run from a clean slate, since the target list has already been modified by the other cases.  How can we reset it?

Reply:
Do something like 

if target state is not empty() -> clear the target_map() 

Maybe its worth creating an internal clear_targets() call that unregisters everything as part of this test as well.

Reply:
You can use `unregister_target("ref")`

Reply:
Turns out that can't be done.  The target is created during program initialization and unregistered during program termination.  In fact, you can't call `unregister_target` at all for a real target because `unregister_target` will throw an exception at termination.

I think I should just say that how the target map gets cleaned up is not the problem of the mutex locks.

Reply:
> for a real target because unregister_target will throw an exception at termination. 

Yeah. Missed that. Thanks.